pkgs/games/prismlauncher/wrapper.nix

```diff
@@ -0,0 +1,75 @@
+{ lib
+, stdenv
+, symlinkJoin
+, prismlauncher-unwrapped
+, wrapQtAppsHook
+, qtbase  # needed for wrapQtAppsHook
+, qtsvg
+, qtwayland
+, xorg
+, libpulseaudio
+, libGL
+, glfw
+, openal
+, jdk8
+, jdk17
+, gamemode
+
+, msaClientID ? null
+, gamemodeSupport ? true
+, jdks ? [ jdk17 jdk8 ]
+, additionalLibs ? [ ]
+}:
+let
+  prismlauncherFinal = prismlauncher-unwrapped.override {
+    inherit msaClientID gamemodeSupport;
+  };
+in
+symlinkJoin {
+  name = "prismlauncher-${prismlauncherFinal.version}";
+
+  paths = [ prismlauncherFinal ];
+
+  nativeBuildInputs = [
+    wrapQtAppsHook
+  ];
+
+  buildInputs = [
+    qtbase
+    qtsvg
+  ]
+  ++ lib.optional (lib.versionAtLeast qtbase.version "6") qtwayland;
+
+  postBuild = ''
+    wrapQtAppsHook
```

Comment:
don't we need a dontWrapQtApps to not double wrap?

Reply:
well this will wrap the unwrapped package that already has `dontWrapQtApps = true` so it's fine.

Reply:
Oh the reason why we do this postBuild, is because symlinkJoin doesn't (seem?) to have a fixup stage, so if we don't do this, the wrapper will never run.

Additionally, the hook can only be called once either way.

https://github.com/NixOS/nixpkgs/blob/68c1807f3c523788b5ce58879ec67ed9c43022d6/pkgs/development/libraries/qt-6/hooks/wrap-qt-apps-hook.sh#L73-L75